lib/services/android-project-service.ts

```diff
@@ -118,19 +119,135 @@ class AndroidProjectService implements IPlatformProjectService {
 			return assetsDirectory;
 
 		}).future<string>()();
-	}
+    }
+
+    private updateMetadata(projectRoot: string): void {
+        var projMetadataDir = path.join(projectRoot, "assets", "metadata");
+        var libsmetadataDir = path.join(projectRoot, "../../lib", this.platformData.normalizedPlatformName, "__metadata");
+        shell.cp("-f", path.join(libsmetadataDir, "*.dat"), projMetadataDir);
+    }
+
+    private generateMetadata(projectRoot: string): void {
+        var metadataGeneratorPath = path.join(__dirname, "../../resources/tools/metadata-generator.jar");
+        var libsFolder = path.join(projectRoot, "../../lib", this.platformData.normalizedPlatformName);
+        var metadataDirName = "__metadata";
+        var inputDir = libsFolder;
+        var outDir = path.join(libsFolder, metadataDirName);
+        this.$fs.ensureDirectoryExists(outDir).wait();
+
+        shell.cp("-f", path.join(__dirname, "../../resources/tools/android.jar"), inputDir);
+        shell.cp("-f", path.join(__dirname, "../../resources/tools/android-support-v4.jar"), inputDir);
+        shell.cp("-f", path.join(projectRoot, "libs/*.jar"), inputDir);
+
+        this.spawn('java', ['-jar', metadataGeneratorPath, inputDir, outDir]).wait();
+    }
 
 	public buildProject(projectRoot: string): IFuture<void> {
-		return (() => {
+        return (() => {
 			var buildConfiguration = options.release ? "release" : "debug";
-			var args = this.getAntArgs(buildConfiguration, projectRoot);
-			this.spawn('ant', args).wait();
+            var args = this.getAntArgs(buildConfiguration, projectRoot);
+            var args2 = this.getAntArgs(buildConfiguration, projectRoot);
+            this.spawn('ant', args).wait();
+            this.generateMetadata(projectRoot);
+            this.updateMetadata(projectRoot);
+            // build the project again in order to include the newly generated metadata
+            this.spawn('ant', args2).wait();
 		}).future<void>()();
 	}
 
 	public isPlatformPrepared(projectRoot: string): IFuture<boolean> {
 		return this.$fs.exists(path.join(projectRoot, "assets", constants.APP_FOLDER_NAME));
-	}
+    }
+
+    private generateBuildFile(projDir: string, targetSdk: string): void {
+        this.$logger.info("Generate build.xml for %s", projDir);
+        var cmd = util.format("android update project -p %s --target %s --subprojects", projDir, targetSdk);
+        this.$childProcess.exec(cmd).wait();
+    }
+
+    private parseProjectProrperies(projDir: string, destDir: string): void {
+
+        var projProp = path.join(projDir, "project.properties");
+
+        if (!this.$fs.exists(projProp).wait()) {
+            this.$logger.warn("File %s does not exist", projProp);
+            return;
+        }
+
+        var lines = fs.readFileSync(projProp, { encoding: "utf-8" }).split("\n");
```

Comment:
but this is valid only in case where projProp file is created on customer's machine. In case we create this file on Linux/Mac machine, disregard my comment